common/flatpak-run.c

```diff
@@ -3165,6 +3165,29 @@ flatpak_run_app (const char     *app_ref,
       g_snprintf (pid_str, sizeof (pid_str), "%d", child_pid);
       pid_path = g_build_filename (instance_id_host_dir, "pid", NULL);
       g_file_set_contents (pid_path, pid_str, -1, NULL);
+
+      g_clear_pointer (&bwrap, flatpak_bwrap_free); // this drops our lock
+
```

Comment:
At this point some other process doing "flatpak run foo" could race and clear the instance though, could it not?

Reply:
but that is already the case anyway... a 'lock snatcher' can sneak in anytime between flatpak creating the directory and bubblewrap taking the lock, and just set up a trap with F_SETLKW, and get the lock as soon as we drop it, no matter how small the window. Or not ?

Reply:
do we know the order in which bubblewrap does things ?

Reply:
also, isn't this error prone anyway ? I mean, we are holding the lock over the spawn_async call - couldn't bwrap race ahead, and try (and fail) to get the lock before spawn async ever returns ?